### DISCUSSION TITLE
Input Fields Inside Dialog

### DISCUSSION THREAD

**Discussion by a-patel02** [shadcn-ui/ui#3069] [Q&A]
Input fields inside dialogs stay focused until first key on dialog open, then lose focus and then stay in focus once reclicked. 

Attaching video for example. As you can see I open dialog, it selects the input and then on the first key press it deselects it. 

any fixes would be helpful, 

thank you

https://github.com/shadcn-ui/ui/assets/89111965/9e1d7d25-950f-4817-8a1d-be31a83338cb

**Comment by pablotanner** [comment]
This likely has to do with how/where you are storing the input field states. Try placing them in the same component where you are also returning the modal. If you share the code I could be more exact

**Comment by Greyn1** [comment]
@a-patel02 Any update on this? Can u let me how did you resolve this? Facing a similar problem.

**Reply by Mastrangelis** [reply]
I had the same issue. The following did the trick
 
```jsx
<DialogContent onOpenAutoFocus={(e) => e.preventDefault()}>
...
</DialogContent />
```

**Reply by erikwibowo** [reply]
this is not working

**Comment by pedroOrvalho** [comment]
Any updates on this?

@a-patel02 are you using Form as well?

im having the same issue but wonly happens if i use form inside the dialog. And input inside form.

If i dont use the form, just the input, i have not issue with focus

**Reply by MarceloGouveia77** [reply]
i got this same error, in my case the problem was:

i had this dynamic input inside my dialog:

```typescript
{offersUuid.map((uuid, index) => (
          <Input
            key={index}
            onChange={(e) => {
              const value = e.target.value;
              setOffersUuid((prev) => {
                const newOffersUuid = [...prev];
                newOffersUuid[index] = value;
                return newOffersUuid;
              });
            }}
          />
        ))}
```

before, the key of input were "uuid", and every onClick event this uuid changed, causing the re-render on every input change, so i changed the key to index and solved.

**Comment by devthub** [comment]
Same here. Is anyone has a workaround?

**Comment by unclebay143** [comment]
Placing your form outside the dialog component should resolve the issue. If your input triggers a re-render, the problem will persist.

**Reply by pedroOrvalho** [reply]
That's not really a solution. Just a concession

**Reply by unclebay143** [reply]
But it sure unblocks, got a solution @pedroOrvalho?

**Comment by SpBills** [comment]
I am running into a similar problem, so I'll put it here for documentation purposes. We have a form inside of a dialog, and every time we do a tab event (even with tabindex="-1" on the dialog container) the form regularly steals our focus.

**Reply by dmausttruefit** [reply]
@SpBills I am curious if you ever found a solution to this?

**Reply by SpBills** [reply]
Sadly not. We're just doing a redesign for the client. They are not happy with losing their tab ability, so this is a better solution.
